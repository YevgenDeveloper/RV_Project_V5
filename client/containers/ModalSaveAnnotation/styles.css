@import 'client/styles/variables';

.image {
  width: 100%;
  min-width: 320px;
  max-width: 640px;
  height: auto;
}

.buttons {
  margin: 30px 0 0;
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
}

.button {
  display: block;
  height: 40px;
  min-width: 100px;
}
