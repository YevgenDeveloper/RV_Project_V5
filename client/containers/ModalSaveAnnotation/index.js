/**
 * App > Containers > ModalSaveAnnotation
 */
import React from 'react';

import { Modal, Button } from 'client/components';
import styles from './styles.css';

type Props = {
  image: string,
  onClickSave: void => void,
  close: void => void,
};

const ModalSaveAnnotation = (props: Props) => (
  <Modal close={props.close}>
    <h1>Save annotation</h1>
    <img alt="" className={styles.image} src={props.image} />
    <div className={styles.buttons}>
      <Button
        label="Cancel"
        onClick={() => props.close()}
        className={styles.button}
      />
      <Button
        blue
        label="Save"
        onClick={() => props.onClickSave()}
        className={styles.button}
      />
    </div>
  </Modal>
);

export default ModalSaveAnnotation;
