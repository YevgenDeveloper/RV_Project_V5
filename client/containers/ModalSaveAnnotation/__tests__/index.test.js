// @flow

import type { ShallowWrapper } from 'enzyme';

import React from 'react';
import { shallow } from 'enzyme';
import ModalSaveAnnotation from '..';

it('renders correctly', () => {
  const wrapper: ShallowWrapper = shallow(
    <ModalSaveAnnotation
      image="Test"
      onClickSave={jest.fn()}
      close={jest.fn()}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
