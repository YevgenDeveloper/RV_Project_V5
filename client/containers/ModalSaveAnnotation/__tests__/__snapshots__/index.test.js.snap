// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalSaveAnnotation
    close={[MockFunction]}
    image="Test"
    onClickSave={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <h1>
          Save annotation
        </h1>,
        <img
          alt=""
          className="image"
          src="Test"
        />,
        <div
          className="buttons"
        >
          <Button
            blue={false}
            className="button"
            dark={false}
            disabled={false}
            label="Cancel"
            onClick={[Function]}
          />
          <Button
            blue={true}
            className="button"
            dark={false}
            disabled={false}
            label="Save"
            onClick={[Function]}
          />
        </div>,
      ],
      "className": undefined,
      "close": [MockFunction],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Save annotation",
        },
        "ref": null,
        "rendered": "Save annotation",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "className": "image",
          "src": "Test",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              blue={false}
              className="button"
              dark={false}
              disabled={false}
              label="Cancel"
              onClick={[Function]}
            />,
            <Button
              blue={true}
              className="button"
              dark={false}
              disabled={false}
              label="Save"
              onClick={[Function]}
            />,
          ],
          "className": "buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "blue": false,
              "className": "button",
              "dark": false,
              "disabled": false,
              "label": "Cancel",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "blue": true,
              "className": "button",
              "dark": false,
              "disabled": false,
              "label": "Save",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h1>
            Save annotation
          </h1>,
          <img
            alt=""
            className="image"
            src="Test"
          />,
          <div
            className="buttons"
          >
            <Button
              blue={false}
              className="button"
              dark={false}
              disabled={false}
              label="Cancel"
              onClick={[Function]}
            />
            <Button
              blue={true}
              className="button"
              dark={false}
              disabled={false}
              label="Save"
              onClick={[Function]}
            />
          </div>,
        ],
        "className": undefined,
        "close": [MockFunction],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Save annotation",
          },
          "ref": null,
          "rendered": "Save annotation",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "image",
            "src": "Test",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Button
                blue={false}
                className="button"
                dark={false}
                disabled={false}
                label="Cancel"
                onClick={[Function]}
              />,
              <Button
                blue={true}
                className="button"
                dark={false}
                disabled={false}
                label="Save"
                onClick={[Function]}
              />,
            ],
            "className": "buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "blue": false,
                "className": "button",
                "dark": false,
                "disabled": false,
                "label": "Cancel",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "blue": true,
                "className": "button",
                "dark": false,
                "disabled": false,
                "label": "Save",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
